Fixing issue 109 - Allow requests with empty body

jefferson-lima opened the pull request:
Here's a fix for the problem described in the [issue 109](https://github.com/alecsammon/php-raml-parser/issues/109).

martin-georgiev commented on src/Validator/ValidatorSchemaHelper.php:
```diff
@@ -127,20 +128,24 @@ public function getResponseHeaders($method, $path, $statusCode, $requiredOnly =
+
+        } catch(EmptyBodyException $e) {
+            throw new EmptyBodyException($e->getMessage());
+        }catch (\Exception $exception) {
```
Could you fix the spacing here and on line 139?

martin-georgiev commented on src/Method.php:
```diff
@@ -13,7 +15,7 @@ class Method implements ArrayInstantiationInterface, MessageSchemaInterface
      * @var array
      */
-    public static $validMethods = ['GET', 'HEAD', 'POST', 'PUT', 'DELETE', 'PATCH'];
+    public static $validMethods = ['GET', 'HEAD', 'POST', 'PUT', 'DELETE', 'PATCH', 'OPTIONS'];
```
Could you add a unit test with `OPTIONS` as method?

martin-georgiev commented on src/Exception/EmptyBodyException.php:
```diff
@@ -0,0 +1,13 @@
+<?php
+namespace Raml\Exception;
+
+
```
Please remove the extra new line here.

martin-georgiev commented on src/Method.php:
```diff
@@ -381,8 +383,12 @@ public function addQueryParameter(NamedParameter $queryParameter)
      */
     public function getBodyByType($type)
     {
+        if(empty($this->getBodies())){
```
Please apply the correct spacing style.

martin-georgiev commented on src/Validator/RequestValidator.php:
```diff
@@ -104,10 +105,24 @@ private function assertValidBody(RequestInterface $request)
+            $schemaBody = $this->schemaHelper->getRequestBody($method, $path, $contentType);
             $schemaBody->getSchema()->validate($body);
+
+        }catch (EmptyBodyException $exception) {
```
The spacing style is slightly off in this newly added line.

jefferson-lima commented:
Thanks for the review, I will be making the corrections soon.

martin-georgiev commented:
@jefferson-lima would you be able to address the code review changes?
There is a momentum for moving the repository to RAML.org and it will be nice to clean up the pile of  MR before that. Thank you!

martin-georgiev commented on src/Validator/ValidatorSchemaHelper.php:
```diff
@@ -142,20 +143,24 @@ public function getResponseHeaders($method, $path, $statusCode, $requiredOnly =
-        } catch (Exception $exception) {
+
+        } catch(EmptyBodyException $e) {
+            throw new EmptyBodyException($e->getMessage());
```
It doesn't look right that the exception is throwing a new instance of the same exception here (`EmptyBodyException`).

jefferson-lima replied:
I believe that what's not quite right is catching an Exception in such a low level of the code, instead of catching a more specific exception. In this case, I need the EmptyBodyException to be propagated, but if it weren't for that piece of code, it would be caught and rethrown as a ValidatorSchemaException. I understand that this is a workaround and it's not straightforward.

jefferson-lima commented:
Hi @martin-georgiev, I believe I fixed the issues you pointed out. Would you please give it another look? Note that I squashed my previous commits into a single one to don't mess up the history, I hope this is not a problem.
